services/api/src/resources/billing/helpers.ts

```diff
@@ -56,11 +64,90 @@ const availabilityFilterFn = filterKey => ({ availability }) =>
  * @param {string} availability High or Standard
  * @param {string} currency the currency
  *
- * @return {object} An object includeing all availability costs
+ * @return {BillingGroupCosts} An object includeing all availability costs
  */
-export const availabiltyProjectsCosts = (projects, availability, currency) => {
+export const availabiltyProjectsCosts = (
+  projects,
+  availability,
+  currency,
+  modifiers: [BillingModifier]
+) => {
   const filteredProjects = projects.filter(availabilityFilterFn(availability));
-  return filteredProjects.length > 0
-    ? getProjectsCosts(currency, filteredProjects)
-    : {};
+  return (filteredProjects.length > 0
+    ? getProjectsCosts(currency, filteredProjects, modifiers)
+    : {}) as BillingGroupCosts;
+};
+
+export const handleAddBillingModifier = async (modifier: BillingModifier) => {
+  const sqlClient = getSqlClient(USE_SINGLETON);
+  const {
+    info: { insertId }
+  } = await query(sqlClient, Sql.addBillingModifier(modifier));
+  const rows = await query(
+    sqlClient,
+    Sql.selectBillingModifier(parseInt(insertId, 10))
+  );
+  return R.prop(0, rows) as BillingModifier;
+};
+
+export const handleGetBillingGroupModifiers = async (
+  gid: string,
+  month: string
+) => {
+  const sqlClient = getSqlClient(USE_SINGLETON);
+
+  const YEAR_MONTH = 'YYYY-MM-DD HH:mm:ss';
+  const monthStart = month
+    ? moment(new Date(month).toISOString()).startOf('month').format(YEAR_MONTH).toString()
+    : undefined;
+  const monthEnd = month
+    ? moment(new Date(month).toISOString()).endOf('month').format(YEAR_MONTH).toString()
+    : undefined;
+
+  const sql = Sql.getAllBillingModifierByBillingGroup(
+    gid,
+    monthStart,
+    monthEnd
+  );
+  const rows = await query(sqlClient, sql);
+  return rows as [BillingModifier];
+};
+
+export const handleUpdateBillingModifier = async (
+  id: number,
+  patch: BillingModifier
+) => {
+  const sqlClient = getSqlClient(USE_SINGLETON);
+  if (!R.isEmpty(patch)) {
+    await query(sqlClient, Sql.updateBillingModifier(id, patch));
+  }
+  const rows = await query(sqlClient, Sql.selectBillingModifier(id));
+  if (rows.length === 0) {
+    throw new Error('Billing modifier does not exist.');
+  }
+  return R.prop(0, rows) as BillingModifier;
+};
+
+export const handleDeleteBillingModifier = async (id: number) => {
+  const sqlClient = getSqlClient(USE_SINGLETON);
+  await query(sqlClient, Sql.deleteBillingModifier(id));
+  return 'success';
+};
+
+export const handleDeleteAllBillingGroupModifier = async (gid: string) => {
+  const sqlClient = getSqlClient(USE_SINGLETON);
+  const sql = Sql.deleteAllBillingModifiersByBillingGroup(gid);
+  await query(sqlClient, sql);
+  return 'success';
+}
+
+export default {
+  extractMonthYear,
+  projectWithBillingDataFn,
+  getProjectsData,
+  availabiltyProjectsCosts,
+  handleAddBillingModifier,
```

Comment:
the helpers.js files for other resources are all what I consider poor models. Since all the handle* functions here are only used in the billingModifier model, lets just inline these and get rid of helper files eventually.

Reply:
I hear what you're saying, definitely makes sense. In going back and looking at the "handler" functions now, that all feels a bit odd and overly abstracted. 

When I see "model" files, I think more MVC though and we don't really have any consistent controllers. 

I'm fine with inlining these, but let's talk more putting an aspirational design pattern in practice and in the developer docs.

Reply:
They're called models because that's what the apollo docs called them, not related to MVC. Of course, now the docs are updated and they don't use that name anymore 🤷

Let's have a chat about what problem I'm trying to solve with the model concept and that probably clears up any questions